### DISCUSSION TITLE
Nitrokey HSM2 fail on windows x64 with OpenSC 0.24.0

### DISCUSSION THREAD

**Discussion by skandersky** [OpenSC/OpenSC#3004] [General]
The nitrokey hsm2 slot cannot be listed with the version 0.24.0
downgraded to version 0.23.0 solve the issue ;

C:\Program Files\OpenSC Project\OpenSC\tools>pkcs11-tool -L
Available slots:
Slot 0 (0x0): Nitrokey Nitrokey HSM 0
  token label        : SmartCard-HSM (UserPIN)
  token manufacturer : www.CardContact.de
  token model        : PKCS#15 emulated
  token flags        : login required, rng, token initialized, PIN initialized
  hardware version   : 24.13
  firmware version   : 3.5
  serial num         : DENK010....
  pin min/max        : 6/15

**Comment by Jakuje** [comment]
Can you provide a debug log from the attempt? is there any difference if you try the nightly builds? There were some recent changes that might have affected this, such as #3006.

**Reply by skandersky** [reply]
here some remarks regarding the version 0.24
When i install the version 0.24 i have a popup with the following 

App: | OpenSC-0.24.0_win64(1).msi
-- | --
Publisher: | Unknown publisher

i don't have this blue window warning with version 0.23 

i accept and run anyway ...

here the output from 0.24 version

C:\Program Files\OpenSC Project\OpenSC\tools>pkcs11-tool.exe -L
Available slots:
Slot 0 (0x0): (GetSlotInfo failed, CKR_DATA_INVALID)

C:\Program Files\OpenSC Project\OpenSC\tools>pkcs11-tool.exe -I
Cryptoki version 3.0
Manufacturer     OpenSC Project
Library          OpenSC smartcard framework (ver 0.24)
error: PKCS11 function C_GetSlotInfo failed: rv = CKR_DATA_INVALID (0x20)
Aborting.

C:\Program Files\OpenSC Project\OpenSC\tools>sc-hsm-tool.exe
Using reader with a card: Nitrokey Nitrokey HSM 0
Version              : 3.5
Config options       :
  User PIN reset with SO-PIN enabled
SO-PIN tries left    : 15
User PIN tries left  : 3

C:\Program Files\OpenSC Project\OpenSC\tools>pkcs11-tool.exe -L
Available slots:
Slot 0 (0x0): (GetSlotInfo failed, CKR_DATA_INVALID)

C:\Program Files\OpenSC Project\OpenSC\tools>pkcs11-tool.exe -I
Cryptoki version 3.0
Manufacturer     OpenSC Project
Library          OpenSC smartcard framework (ver 0.24)
error: PKCS11 function C_GetSlotInfo failed: rv = CKR_DATA_INVALID (0x20)
Aborting.

C:\Program Files\OpenSC Project\OpenSC\tools>sc-hsm-tool.exe
Using reader with a card: Nitrokey Nitrokey HSM 0
Version              : 3.5
Config options       :
  User PIN reset with SO-PIN enabled
SO-PIN tries left    : 15
User PIN tries left  : 3

i don't have those issues on 0.23 when reinstall it ,
[opensc-debug.zip](https://github.com/OpenSC/OpenSC/files/14208275/opensc-debug.zip)

**Reply by Jakuje** [reply]
Can you check if this is resolved with the 0.25.0. If not, we will need an updated debug log.

**Comment by frankmorgner** [comment]
please open an issue (rather than a discussion), if the problem is also present for the current version of OpenSC.
